src/transformers/trainer_callback.py

```diff
@@ -631,7 +632,10 @@ def on_log(self, args, state, control, logs=None, **kwargs):
             # but avoid doing any value pickling.
             shallow_logs = {}
             for k, v in logs.items():
-                shallow_logs[k] = v
+                if isinstance(v, str) and len(v) > self.max_str_len:
+                    shallow_logs[k] = f"[String too long to display, length: {len(v)}]"
```

Comment:
Maybe you can even tell them to modify max_str_len in the callback if they want to increase the length

Reply:
I've added some docstrings and included a hint in the message.

Reply:
Hi @SunMarc, just checking in to see if there's any update on this PR. Please let me know if any additional changes are needed. Thanks!

Reply:
Nice, thanks for your work !